dcnm/resource_dcnm_network.go

```diff
@@ -718,14 +775,26 @@ func resourceDCNMNetworkCreate(d *schema.ResourceData, m interface{}) error {
 	} else {
 		networkProfile.NVEId = ""
 	}
-	networkProfile.NetworkName = name
-	networkProfile.SegmentID = segID
 
-	configStr, err := json.Marshal(networkProfile)
-	if err != nil {
-		return err
+	if networkConfig, ok := d.GetOk("template_props"); ok {
+		networkConfigMap := networkConfig.(map[string]interface{})
```

Comment:
Not sure if we should assign the values in the key `template_props` directly without any further verification. I would argue that users setting the key `template_props` must be completely sure that the JSON payload is built based on those values without further verification

Maybe we should only check if the VLAN ID and the Multicast group are part of the keys. If not we generate them. What are your thoughts?

Reply:
I think having a check to ensure the template_props are verified is a good step to add. Yes, i think VLAN ID and multicast group are a good start. I'll think more about what other parameters are important.